src/terminal/terminal.rs

```diff
@@ -657,6 +669,86 @@ where
         Ok(())
     }
 
+    /// Implement `Self::insert_before` using scrolling regions.
```

Comment:
Note to self, read the code below this point more thoroughly - skimmed instead of reading it deeply.